sectionproperties/analysis/section.py

```diff
@@ -2126,6 +2137,149 @@ def get_sf_p(self):
+        :return: Resultant normal and shear stresses list[(sigma_zz, tau_xz, tau_yz)]. If a point it not in the
+            section then None is returned.
+        :rtype: Union[Tuple[float, float, float], None]
+        """
+        sigs = self.get_stress_at_points(
+            [pt], N, Mxx, Myy, M11, M22, Mzz, Vx, Vy, agg_func
+        )
+        return next(sigs)
+
+    def get_stress_at_points(
+        self,
+        pts: List[List[float]],
+        N=0,
+        Mxx=0,
+        Myy=0,
+        M11=0,
+        M22=0,
+        Mzz=0,
+        Vx=0,
+        Vy=0,
+        agg_func=np.average,
+    ) -> List[Tuple]:
+        """Calaculates the stress at a set of points within an element for given design actions
+        and returns *(sigma_zz, tau_xz, tau_yz)*
+
+        :param pts: The points. A list of several x and y coordinates
+        :type pts: list[list[float, float]]
+        :param float N: Axial force
+        :param float Vx: Shear force acting in the x-direction
+        :param float Vy: Shear force acting in the y-direction
+        :param float Mxx: Bending moment about the centroidal xx-axis
+        :param float Myy: Bending moment about the centroidal yy-axis
+        :param float M11: Bending moment about the centroidal 11-axis
+        :param float M22: Bending moment about the centroidal 22-axis
+        :param float Mzz: Torsion moment about the centroidal zz-axis
+        :param agg_function: A function that aggregates the stresses if the point is shared by several elements.
+            If the point, pt, is shared by several elements (e.g. if it is a node or on an edge), the stress
+            (sigma_zz, tau_xz, tau_yz) are retrieved from each element and combined according to this function.
+            By default, `numpy.average` is used.
+        :type agg_function: function, optional
+        :return: Resultant normal and shear stresses list[(sigma_zz, tau_xz, tau_yz)]. If a point it not in the
+            section then None is returned for that element in the list.
+        :rtype: List[Union[Tuple[float, float, float], None]]
+        """
+
+        action = {
+            "N": N,
+            "Mxx": Mxx,
+            "Myy": Myy,
+            "M11": M11,
+            "M22": M22,
+            "Mzz": Mzz,
+            "Vx": Vx,
+            "Vy": Vy,
+        }
+
+        sect_prop = {
+            "ea": self.section_props.ea,
+            "cx": self.section_props.cx,
+            "cy": self.section_props.cy,
+            "ixx": self.section_props.ixx_c,
+            "iyy": self.section_props.iyy_c,
+            "ixy": self.section_props.ixy_c,
+            "i11": self.section_props.i11_c,
+            "i22": self.section_props.i22_c,
+            "phi": self.section_props.phi,
+            "j": self.section_props.j,
+            "Delta_s": self.section_props.Delta_s,
+            "nu": self.section_props.nu_eff,
+        }
+
+        for pt in pts:
+            query_geom = asPoint(pt)
```

Comment:
I get the following warning:

> sectionproperties/analysis/section.py:2179: ShapelyDeprecationWarning: The proxy geometries (through the 'asShape()', 'asPoint()' or 'PointAdapter()' constructors) are deprecated and will be removed in Shapely 2.0. Use the 'shape()' function or the standard 'Point()' constructor instead.

Reply:
Changing this to ``Point(pt)`` works for me, with ``from shapely.geometry import Point``.

Reply:
Got it, I must be running an older version of Shapely I guess